src/Mapless-SQLite-Core/MaplessSQLitePool.class.st

```diff
@@ -0,0 +1,245 @@
+"
+Connection pool for the Mapless repository using SQLite
+"
+Class {
+	#name : #MaplessSQLitePool,
+	#superclass : #MaplessAbstractPool,
+	#instVars : [
+		'databaseName',
+		'clientsMonitor',
+		'busyClients',
+		'idleClients',
+		'clientsMutex',
+		'maxClients',
+		'databaseFilenameOrKeyword'
+	],
+	#category : #'Mapless-SQLite-Core'
+}
+
+{ #category : #'instance creation' }
+MaplessSQLitePool class >> defaultMaxClients [
+	"Default maximum quantity of clients for the connection pool used in case no max was specified"
+
+	^ 50
+]
+
+{ #category : #'instance creation' }
+MaplessSQLitePool class >> local [
+	^ self memory
+]
+
+{ #category : #'instance creation' }
+MaplessSQLitePool class >> memory [
+	^ self new initializeOn: ':memory:'
+]
+
+{ #category : #'instance creation' }
+MaplessSQLitePool class >> on: aDatabaseFilename [
+	^ self new initializeOn: aDatabaseFilename
+]
+
+{ #category : #accessing }
+MaplessSQLitePool >> busyClients [
+	"Answers the bunch of clients that are currently busy"
+
+	^ busyClients ifNil: [ self initializeBusyClients ]
+]
+
+{ #category : #accessing }
+MaplessSQLitePool >> clientsMonitor [
+	"Answers the monitor used to protect clients usage"
+
+	^ clientsMonitor ifNil: [ self initializeClientsMonitor ]
+]
+
+{ #category : #accessing }
+MaplessSQLitePool >> currentClientDynamicVariable [
```

Comment:
mmh.. wouldn't it be better `currentClientDynamicVariableClass`

Reply:
@fgarau yes it would! great catch